examples/variational_autoencoder.py

```diff
@@ -1,111 +1,205 @@
-'''This script demonstrates how to build a variational autoencoder with Keras.
+'''Example of VAE on MNIST dataset using MLP
 
- #Reference
+This VAE has a modular design. The encoder, decoder and vae
+are 3 models that share weights. After training vae,
+the encoder can be used to  generate latent vectors.
+The decoder can be used to generate MNIST digits by sampling the
+latent vector from a gaussian dist with mean=0 and std=1.
 
- - Auto-Encoding Variational Bayes
-   https://arxiv.org/abs/1312.6114
+[1] Kingma, Diederik P., and Max Welling.
+"Auto-encoding variational bayes."
+arXiv preprint arXiv:1312.6114 (2013).
 '''
-from __future__ import print_function
 
-import numpy as np
-import matplotlib.pyplot as plt
-from scipy.stats import norm
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
 
-from keras.layers import Input, Dense, Lambda
+from keras.layers import Lambda, Input, Dense
 from keras.models import Model
-from keras import backend as K
-from keras import metrics
 from keras.datasets import mnist
+from keras.losses import mse, binary_crossentropy
+from keras.utils import plot_model
+from keras import backend as K
 
-batch_size = 100
-original_dim = 784
-latent_dim = 2
-intermediate_dim = 256
-epochs = 50
-epsilon_std = 1.0
-
-
-x = Input(shape=(original_dim,))
-h = Dense(intermediate_dim, activation='relu')(x)
-z_mean = Dense(latent_dim)(h)
-z_log_var = Dense(latent_dim)(h)
+import numpy as np
+import matplotlib.pyplot as plt
+import argparse
+import os
 
 
+# reparameterization trick
+# instead of sampling from Q(z|X), sample eps = N(0,I)
+# z = z_mean + sqrt(var)*eps
 def sampling(args):
-    z_mean, z_log_var = args
-    epsilon = K.random_normal(shape=(K.shape(z_mean)[0], latent_dim), mean=0.,
-                              stddev=epsilon_std)
-    return z_mean + K.exp(z_log_var / 2) * epsilon
-
-# note that "output_shape" isn't necessary with the TensorFlow backend
-z = Lambda(sampling, output_shape=(latent_dim,))([z_mean, z_log_var])
-
-# we instantiate these layers separately so as to reuse them later
-decoder_h = Dense(intermediate_dim, activation='relu')
-decoder_mean = Dense(original_dim, activation='sigmoid')
-h_decoded = decoder_h(z)
-x_decoded_mean = decoder_mean(h_decoded)
+    """Implements reparameterization trick by sampling
```

Comment:
Please format all docstrings in this CL like other docstrings in the codebase:

```python
"""One-line summary ending in a period.

# Arguments
    argument_1: Description (may include type if relevant).
    argument_2: Description.

# Returns

Description of the output.
"""
```

Reply:
fixed